File: 2018/Case.ApplySysOrient/Entry/CmdMain.cs
using Autodesk.Revit.Attributes;
using Autodesk.Revit.DB;
using Autodesk.Revit.UI;
using Case.ApplySysOrient.Data;
using Case.ApplySysOrient.UI;
using System;

namespace Case.ApplySysOrient.Entry
{

  [Transaction(TransactionMode.Manual)]
  public class CmdMain : IExternalCommand
  {

    /// <summary>
    /// Duct Command
    /// </summary>
    /// <param name="commandData"></param>
    /// <param name="message"></param>
    /// <param name="elements"></param>
    /// <returns></returns>
    public Result Execute(ExternalCommandData commandData, ref string message, ElementSet elements)
    {

      try
      {

        // Version
        if (!commandData.Application.Application.VersionName.Contains("2018"))
        {

          // Failure
          using (TaskDialog td = new TaskDialog("Cannot Continue"))
          {
            td.TitleAutoPrefix = false;
            td.MainInstruction = "Incompatible Revit Version";
            td.MainContent = "";
            td.Show();
          }
          return Result.Failed;

        }

        // Settings
        clsSettings m_s = new clsSettings(commandData);

        // Main Category Selection Form
        using (form_Orient dlg = new form_Orient())
        {
          dlg.ShowDialog();

          if (!dlg.DoConduit && !dlg.DoDuct && !dlg.DoPipe & !dlg.DoTray) return Result.Cancelled;

          // Process Data
          if (dlg.DoConduit) m_s.ProcessConduit();
          if (dlg.DoDuct) m_s.ProcessDuct();
          if (dlg.DoPipe) m_s.ProcessPipe();
          if (dlg.DoTray) m_s.ProcessTray();

        }

        // Success
        return Result.Succeeded;

      }
      catch (Exception ex)
      {

        // Failure
        message = ex.Message;
        return Result.Failed;

      }

    }

  }
}

File: 2018/Case.RoomSync/Case.RoomSync/Entry/CmdMain.vb
Imports Autodesk.Revit.Attributes
Imports Autodesk.Revit.DB
Imports Autodesk.Revit.UI
Imports [Case].RoomSync.Data
Imports [Case].RoomSync.API

Namespace Entry

  <Transaction(TransactionMode.Manual)>
  Public Class CmdMain
    Implements IExternalCommand

    ''' <summary>
    ''' Command
    ''' </summary>
    ''' <param name="commandData"></param>
    ''' <param name="message"></param>
    ''' <param name="elements"></param>
    ''' <returns></returns>
    ''' <remarks></remarks>
    Public Function Execute(ByVal commandData As ExternalCommandData,
                            ByRef message As String,
                            ByVal elements As ElementSet) As Result Implements IExternalCommand.Execute
      Try

        ' Version
        If Not commandData.Application.Application.VersionName.Contains("2018") Then

          Using td As New TaskDialog("Cannot Continue")
            With td
              .TitleAutoPrefix = False
              .MainInstruction = "Incompatible Revit Version"
              .MainContent = ""
              .Show()
            End With
          End Using
          Return Result.Cancelled

        End If

        ' Settings
        Dim m_s As New clsSettings(commandData)

        ' Construct and Display Form
        Using d As New form_Sync(m_s)
          d.ShowDialog()
        End Using

        ' Success
        Return Result.Succeeded

      Catch ex As Exception

        ' Failure
        message = ex.Message
        Return Result.Failed

      End Try

    End Function

  End Class
End Namespace